### PR TITLE
Using black to check code formatting

### PR DESCRIPTION
**Pull request by bkowshik**
Closes: https://github.com/mapbox/robosat/issues/15

## Next actions
- [x] Watch the Travis build fail because of not-blackened code
- [x] Format code with Black and commit the changes
- [x] Watch the Travis build succeed, now with blackened code
- [x] Review and merge

### CODE REVIEW

**Comment by daniel-j-h** [comment]
Looks like black requires Python 3.6+. This makes it hard to run on Ubuntu 16.04 with Python 3.5 by default.

Should we recommend folks using a Docker container for this? Or what's the best way here?

**Comment by bkowshik** [comment]
>  Watch the Travis build fail because of not-blackened code

```bash
All done! 💥 💔 💥
37 files would be reformatted, 5 files would be left unchanged.
The command "black . --exclude .env --check" exited with 1.
Done. Your build exited with 1.
```

**Comment by bkowshik** [comment]
> Format code with Black and commit the changes

```bash
$ black . --exclude .env
...
All done! ✨ 🍰 ✨
37 files reformatted, 5 files left unchanged.
```

> Watch the Travis build succeed, now with blackened code

:tada:

> Should we recommend folks using a Docker container for this? Or what's the best way here?

Don't have any recommendations here. When users see their build fail on Travis because of code formatting with Black, would the worst case be users having to create a new virtual environment (if required) with Python 3.6 and Black and then run automated code formatting.

File: robosat/datasets.py [hunk 1]
```diff
@@ -42,21 +42,27 @@ def __getitem__(self, i):
...
         self.target = SlippyMapTiles(target, target_transform)
 
-        assert len(set([len(dataset) for dataset in self.inputs])) == 1, 'same number of tiles in all inputs'
-        assert len(self.target) == len(self.inputs[0]), 'same number of tiles in inputs and target'
+        assert (
+            len(set([len(dataset) for dataset in self.inputs])) == 1
+        ), "same number of tiles in all inputs"
+        assert len(self.target) == len(
+            self.inputs[0]
+        ), "same number of tiles in inputs and target"
```
Comment (daniel-j-h) [inline robosat/datasets.py]: We should at least adapt the line length to what we specify in the flake config. Or why is it breaking here?

File: robosat/features/core.py [hunk 2]
```diff
@@ -102,38 +102,40 @@ def contours(mask):
       The detected contours as a list of points and the contour hierarchy.
 
     Note: the hierarchy can be used to re-construct polygons with holes as one entity.
-    '''
+    """
 
-    _, contours, hierarchy = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+    _, contours, hierarchy = cv2.findContours(
+        mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE
+    )
```
Comment (daniel-j-h) [inline robosat/features/core.py]: Wow these parenthesis placements are ugly :see_no_evil: Is there a way to change it?

File: robosat/datasets.py (hunk 1 above)
Comment (bkowshik) [reply robosat/datasets.py]: Looks like was from the `--line-length` parameter. Now, set that to 120 and code is formatted as we expect and looks good as well. 😄

File: robosat/features/core.py (hunk 2 above)
Comment (bkowshik) [reply robosat/features/core.py]: This was also fixed with `--line-length 120`

**Comment by bkowshik** [comment]
Rebased this branch with `master` @daniel-j-h let me know if we are good to merge here.

**Comment by daniel-j-h** [comment]
Looks good, can you squash your commits into one, rebase against master, and then merge?

**Pull request merged**